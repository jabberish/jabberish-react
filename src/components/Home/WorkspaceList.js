import React from 'react';
import PropTypes from 'prop-types';

import { Grid, Paper, makeStyles, Typography, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

const useStyles = makeStyles(theme => ({
  paper: {
    height: 140,
    width: 300,
    textDecoration: 'none'
  },
  control: {
    padding: theme.spacing(2),
  },
  typography: {
    textDecoration: 'none'
  },
  link: {
    textDecoration: 'none'
  }
}));

const WorkspaceList = ({ workspaces, updateWorkspace }) => {
  const classes = useStyles();

  return (
    <>
      <h3>Workspace List</h3>
      <Grid container justify="left" spacing={2}>
        {workspaces.map(({ workspace }) => (
          <Grid key={workspace._id} item>
            <Link to="./workspace" onClick={() => updateWorkspace(workspace._id)} className={classes.link}>
              <Paper className={classes.paper} >
                <Typography variant="subtitle1" className={classes.typography}>
                  {workspace.name}
                </Typography>
                <Button style={{ zIndex: 100000 }} onClick={e => e.preventDefault()}>Test</Button>
              </Paper>
            </Link>
          </Grid>
        ))}
      </Grid>
    </>
  );
};

WorkspaceList.propTypes = {
  workspaces: PropTypes.array.isRequired,
  updateWorkspace: PropTypes.func.isRequired
};

export default WorkspaceList;
