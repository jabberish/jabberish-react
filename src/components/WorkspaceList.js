import React from 'react';
import PropTypes from 'prop-types';

const WorkspaceList = ({ workspaces }) => {
  const workspaceList = workspaces.map((workspace, i) => {
    return <li key={i}>{workspace.workspace.name}</li>;
  });

  return (
    <>
      <h3>Workspace List</h3>
      <ul>{workspaceList}</ul>
    </>
  );
};

WorkspaceList.propTypes = {
  workspaces: PropTypes.array.isRequired
};

export default WorkspaceList;
