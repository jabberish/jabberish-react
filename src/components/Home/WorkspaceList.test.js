import React from 'react';
import { shallow } from 'enzyme';
import WorkspaceList from './WorkspaceList';

describe('WorkspaceList component', () => {
  it('renders WorkspaceList', () => {
    const wrapper = shallow(<WorkspaceList 
      workspaces={[]} 
      updateWorkspace={() => {}} 
      handleOpenDialog={() => {}} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
