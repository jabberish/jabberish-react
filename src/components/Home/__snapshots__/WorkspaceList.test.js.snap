// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceList component renders WorkspaceList 1`] = `
<Fragment>
  <h3>
    Workspace List
  </h3>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-container-6"
    container={true}
    justify="flex-start"
    spacing={2}
  >
    <CreateWorkspace
      handleOpenDialog={[Function]}
    />
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
